apps/site/src/pages/settings/billing-settings-panel/tax-id-section.tsx

```diff
@@ -0,0 +1,200 @@
+import { StripeTaxId } from "@local/internal-api-client";
+import { Close } from "@mui/icons-material";
+import {
+  Autocomplete,
+  Box,
+  CircularProgress,
+  Collapse,
+  Fade,
+  IconButton,
+  outlinedInputClasses,
+  Typography,
+} from "@mui/material";
+import { AxiosError } from "axios";
+import debounce from "lodash/debounce";
+import {
+  FunctionComponent,
+  ReactNode,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+
+import { TextField } from "../../../components/text-field";
+import { internalApi } from "../../../lib/internal-api-client";
+import { StripeTaxIdType, stripeTaxIdTypes } from "./tax-id-types";
+
+export const TaxIdSection: FunctionComponent = () => {
+  const [existingTaxId, setExistingTaxId] = useState<StripeTaxId | null>();
+
+  const taxIdInputRef = useRef<HTMLInputElement>(null);
+  const [taxIdValue, setTaxIdValue] = useState<string>("");
+  const [errorMessage, setErrorMessage] = useState<ReactNode>();
+  const [loading, setLoading] = useState<boolean>();
+  const [stripeTaxIdType, setStripeTaxIdType] =
+    useState<StripeTaxIdType | null>(null);
+
+  const fetchTaxId = useCallback(async () => {
```

Comment:
This is only used inside an effect, it could just be inside that function

Reply:
Yes, but because it's async it'd require something like `void (async () => {})()`, is that still preferable?

Reply:
Ah, true. It's fine as it is